pkg/rke2/rke2.go

```diff
@@ -166,5 +175,150 @@ func setup(clx *cli.Context, cfg Config) error {
 	}
 	executor.Set(&sp)
 
+	disabledItems := map[string]bool{
+		"kube-apiserver":          disableAPIServer,
+		"kube-scheduler":          disableScheduler,
+		"kube-controller-manager": disableControllerManager,
+		"etcd":                    disableETCD,
+	}
+	return removeOldPodManifests(dataDir, disabledItems)
+}
+
+func podManifestsDir(dataDir string) string {
+	return filepath.Join(dataDir, "agent", config.DefaultPodManifestPath)
+}
+
+func removeOldPodManifests(dataDir string, disabledItems map[string]bool) error {
+	var kubeletStandAlone bool
+
+	kubeletErr := make(chan error)
+	containerdErr := make(chan error)
+
+	ctx, cancel := context.WithTimeout(context.Background(), (5 * time.Minute))
+	defer cancel()
+
+	manifestDir := podManifestsDir(dataDir)
+	for component, disabled := range disabledItems {
+		if disabled {
+			manifestName := filepath.Join(manifestDir, component+".yaml")
+			if _, err := os.Stat(manifestName); err == nil {
+				kubeletStandAlone = true
+				if err := os.Remove(manifestName); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	if kubeletStandAlone {
+		kubeletCmd := exec.Command("kubelet")
+		containerdCmd := exec.Command("containerd")
```

Comment:
Use CommandContext, it will kill the processes for you when the context is cancelled.
```suggestion
		kubeletCmd := exec.CommandContext(ctx, "kubelet")
		containerdCmd := exec.CommandContext(ctx, "containerd")
```